merkle/objhasher/objhasher.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objhasher
+
+import (
+	"crypto/sha256" // Use SHA256 to match ObjectHash.
+
+	"github.com/benlaurie/objecthash/go/objecthash"
+)
+
+// ObjectHasher uses ObjectHash to compute leaf hashes.
+var ObjectHasher = objhasher{}
+
+// ObjectHash does not use `1` as any of its type prefixes,
+// preserving domain separation.
+const nodeHashPrefix = 1
+
+type objhasher struct{}
+
+// HashEmpty returns the hash of an empty element for the tree
+func (t objhasher) HashEmpty() []byte {
+	return sha256.New().Sum(nil)
```

Comment:
I think the slight performance penalty is worth the self documentation of using the raw computation.  

This doesn't have a prefix because it matches the spec for rfc6962.